### ISSUE TITLE
Reproducibility issue caused by joblib

### ISSUE THREAD

**Issue by yvchao** [vanderschaarlab/autoprognosis#15]
#### Describe the bug
Outputs of some predictor plugins (e.g., random forest) vary at different runs regardless of the random seed set in the environment.

#### Example to reproduce
```Python
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
import pandas as pd
import numpy as np
import random
from pathlib import Path
import warnings

warnings.filterwarnings('ignore')

from adjutorium.studies.classifiers import ClassifierStudy
from adjutorium.utils.serialization import load_model_from_file, save_model_to_file
from adjutorium.utils.tester import evaluate_estimator

X, Y = load_breast_cancer(return_X_y=True, as_frame=True)

data = X.copy()
data["target"] = Y

workspace = Path("workspace")
workspace.mkdir(parents=True, exist_ok=True)

study_name = "classification_example"

study = ClassifierStudy(
    study_name = study_name,
    dataset=data, # pandas DataFrame
    target="target", # the label column in the dataset
    num_iter=1, # how many trials to do for each candidate. Default: 50
    num_study_iter=1, # how many outer iterations to do. Default: 5
    timeout=1, # timeout for optimization for each classfier. Default: 600 seconds
    classifiers=["random_forest"],
    workspace=workspace,
    score_threshold=0.49
)

study.run()

def reproducibility_check(seed=0):
    # set random seed in the outside environment
    random.seed(seed)
    np.random.seed(seed)
    output = workspace / study_name / "model.p"

    model = load_model_from_file(output)

    metrics = evaluate_estimator(model, X, Y,seed=seed)
    print(model.name())
    print(f"score: {metrics['str']}")

# we expect the output to be the same.
reproducibility_check(seed=0)
reproducibility_check(seed=0)
```
#### Result
The outputs from the two runs with the same random seed 0 are different.
```text
0.9999999900000002 * (random_forest)
score: {'aucroc': '0.9902 +/- 0.0061'}
0.9999999900000002 * (random_forest)
score: {'aucroc': '0.9911 +/- 0.0051'}
```

#### Expected behavior
The outputs from different runs should be the same when the random seed is fixed in the environment.

#### Note
This behavior is only observed for weighted ensembles. I suppose it is related to the reproducibility issue when using joblib:
https://github.com/vanderschaarlab/adjutorium-framework/blob/3bcb068a71ccd6261cf0a77238f56312a61a2f72/src/adjutorium/plugins/ensemble/classifiers.py#L116

See [Random state within joblib.Parallel](https://joblib.readthedocs.io/en/latest/auto_examples/parallel_random_state.html).

#### Version
Latest version on GitHub.
